src/gui/DatabaseOpenWidget.cpp

```diff
@@ -148,7 +148,7 @@ void DatabaseOpenWidget::load(const QString& filename)
     m_filename = filename;
     m_ui->fileNameLabel->setRawText(m_filename);
 
-    m_ui->comboKeyFile->addItem(tr("Select file..."), -1);
+    m_ui->comboKeyFile->addItem(tr("Select key file (if any)..."), -1);
```

Comment:
I don't like the (if any) part. It doesn't reduce confusion IMO

Reply:
Yeah, I don't really like it either, but I thought that writing only "Select key file..." will not be explicit enough and people will still try to put their KDBX in this field.

Reply:
Would it be too extreme to have all of this hidden behind an "Advanced" arrow that has its state immediately saved to settings file so it persists?

Reply:
Then just write (this is not your database file)

Reply:
Or actually because that's likely not a native UI element, a checkbox with "[ ] Use additional security tokens" which acts as a toggle to show/hide the fields?

Reply:
I don't want to make it too long. Too much text clutter the whole layout too much.

Reply:
The rejection of kdbx files is sufficient

Reply:
It's kinda after-the-fact, though. I prefer UIs which do not even tempt the user to do the wrong thing.

Reply:
IMO it would be better for "Enter Additional Credentials" line to read "Enter Additional Credentials (Optional)" or "Enter Additional Credentials (if any)" so that it's clearer that both key files/hardware keys aren't required unless set by the user.

Reply:
Or (Advanced Users)

Reply:
Adding "(Advanced Users)" is not encouraging inexperienced users to use good practices, i.e., use a keyfile. It puts them off. Even from thinking about using a key file ("I'll never be an advanced user anyways").

I don't like the whole line "Enter Additional Credentials" and I'd remove it completely. But that might be a very personal view and has not too much to do with the issue at hand.

How a about a tooltip-like box, which appears as soon as and while the cursor hovers or moves over the "Browse" button? It would have plenty space to convey to the inexperienced user whether or not a key file must be specified. It would not clutter the unlock dialog GUI layout. It will disrupt inexperienced, cautious users, which it should. Experienced users will soon learn to ignore the box, if they are not saving paths to recent databases and key files anyways, in which case they will hover the "Browse" button only once and (basically) never see the box again.
![Screenshot from 2019-11-07 09-47-59](https://user-images.githubusercontent.com/3578416/68374126-90628100-0144-11ea-9011-790954d2d76f.png)

Reply:
I wouldn't add a tooltip to the browse button. It's too unreliable to trigger and too easily dismissed or overlooked.

I took a few of the suggestions in this thread and combined them into a slightly different solution:

![image](https://user-images.githubusercontent.com/911270/68374791-d1a76080-0145-11ea-97b8-abc819bbd3fa.png)

TODO: create the key file FAQ entry that opens when the user clicks the help icon.

Reply:
Note: I did not title-case the (if any), because it looks too ugly.

Reply:
For reference I imagined a single checkbox that remembers state between startups to toggle the extra fields:

![image](https://user-images.githubusercontent.com/106598/68396963-18fc1400-017b-11ea-98fb-f34b4b670603.png)

As discussed on IRC this is unlikely to be implemented but I wanted it to be part of this thread.